### PR TITLE
refactor: upgrade go version to 1.18.6

### PR DESCRIPTION
**Pull request by rajatgupta24**
This PR updates the go to `1.18.6`.
And the `linter.sh` need to be updated.

related to: https://github.com/jenkins-x/jx/issues/8283

- [x] Change the version in `go.mod` file.
- [x] Run `go mod tidy` and ensure that it does not error out.
- [x] Change `GO_VERSION` in `makefile` to the version you are upgrading.
- [x] Check if there are any references to an older version and replace those.
- [x] Verify that
  - [x] The build works by running `make build`.
  - [x] Tests are working, by running `make test`.
  - [x] Linting checks are working by running `make lint`.

### CODE REVIEW

**Review by ankitm123 — APPROVED** [review]

**Comment by ankitm123** [comment]
Modify the owners file (check jx-pipeline for reference) and keep vbehar in the owners list.

**Comment by vbehar** [comment]
> Modify the owners file (check jx-pipeline for reference) and keep vbehar in the owners list.

thanks ;-)

**Review by vbehar — APPROVED** [review]

File: OWNERS [hunk 1]
```diff
@@ -1,8 +1,6 @@
 approvers:
-- rawlingsj
-- jstrachan
+- maintainers
```
Comment (rajatgupta24) [inline OWNERS:2]: Is this good?
I also added `OWNER_ALIASES`

**Review by ankitm123 — APPROVED** [review]

**Pull request merged**